swan-lake/learn/testing-ballerina-code/writing-tests.md

```diff
@@ -214,6 +214,55 @@ function intAdd(int a, int b) returns (int) {
 }
 ```
 
+### **assertExactEquals(Any actual, Any expected, string message)**
+
+Asserts that the actual is exactly equal to the expected with an optional message.
+
+***Example:***
+```ballerina
+import ballerina/test;
+
+type Person object {
+    public string name = "";
+    public int age = 0;
+    public Person? parent = ();
+    private string email = "default@abc.com";
+    string address = "No 20, Palm grove";
+};
+
+@test:Config {}
+function testAssertExactEqualsObject() {
+    Person p1 = new;
+    Person p2 = p1;
+    test:assertExactEquals(p1, p2, msg = "Objects are not exactly equal");
+}
+```
+
+### **assertExactNotEquals(Any actual, Any expected, string message)**
+
+Asserts that the actual is not exactly equal to the expected with an optional message.
```

Comment:
```suggestion
Asserts that the actual value is not exactly equal to the expected value with an optional message.
```

Reply:
@praneesha it's not only the value, it's the complete equality(including storage identity) that is being checked with this function.

Reply:
We have to have some noun here. Can we say, the "exact assertion"?